docs-src/api-overview.md

```diff
@@ -79,6 +96,132 @@ BackendPolicy. For more information on what may be configured with this resource
+> proxy their application to the outside world. Team C deploys their own Gateway 
+> “dedicated-gw”  in the “C” Namespace that can only be used by apps in the "C" 
+> Namespace.
+
+<!-- source: https://docs.google.com/presentation/d/1neBkFDTZ__vRoDXIWvAcxk2Pb7-evdBT6ykw_frf9QQ/edit?usp=sharing -->
+![route binding](images/gateway-route-binding.png)
+
+There is a lot of flexibility in how Routes can bind to Gateways to achieve
+different organizational policies and scopes of responsibility. These are 
+different relationships that Gateways and Routes can have:
+
+- **One-to-one** - A Gateway and Route may be deployed and used by a single 
+  owner and have a one-to-one relationship. Team C is an example of this. 
+- **One-to-many** - A Gateway can have many Routes bound to it that are owned by
+  different teams from across different Namespaces. Teams A and B are an example 
+  of this.
+- **Many-to-one** - Routes can also be bound to more than one Gateway, allowing 
+  a single Route to control application exposure simultaneously across different
+  IPs, load balancers, or networks.
+
+*In summary, Gateways select Routes and Routes also select Gateways. When both 
+select each otherthen a Route will bind to a Gateway. When Routes are bound to a
+Gateway it means their collective routing rules are configured on the
+underlying load balancers or proxies that are managed by that Gateway. Thus, a
+Gateway is a logical representation of a networking data plane that can be
+configured through Routes.*
+
+#### Route Selection
+
+A Gateway selects routes based on the Route metadata, specifically the kind,
+namespace, and labels of Route resources. Routes are actually bound to specific
+listeners within the Gateway so each listener has a `listener.routes` field
+which selects Routes by one or more of the following criterea:
+
+- **Label** - A Gateway can select Routes via labels that exist on the 
+resource (similar to how Services select Pods via Pod labels).
+- **Kind** - A Gateway listener can only select a single type of Route 
+resource. This could be an HTTPRoute, TCPRoute, or a custom Route type.
+- **Namespace** - A Gateway can also control from which Namespaces Routes can be
+selected via the `namespaces.from` field. It supports three possible values:
+    - `SameNamespace` is the default option. Only Routes in the same namespace 
+    as this Gateway will be selected. 
+    - `All` will select Routes from all Namespaces.
+    - `Selector` means that Routes from a subset of Namespaces selected by a 
+    Namespace label selector will be selected by this Gateway. When `Selector` 
+    is used then the `listeners.routes.namespaces.selector` field can be used to 
+    specify label selectors. This field is not supported with `All` or 
+    `SameNamespace`.
+
+The below Gateway will select all HTTPRoute resources with the `gateway:
+prod-web-gw` across all Namespaces in the cluster.
```

Comment:
I think using the `gateway` label in this example could add further confusion. Maybe instead we could use a common k8s label like `app: web` and/or `traffic: external`.

Reply:
There's two ways of going about this ...
- Binding via Gateway symbolic reference like `expose=prod-web-gw`. This feels like it would be useful for most kinds of deployments. Using a single key for Gateway selection does not work so well if Routes must be exposed by multiple Gateways though because you can't use the key more than once.
- Binding via Gateway characteristics like `network=internet && stage=prod`. This feels like it would be useful if you had many GatewayClasses and so you wanted to make it easier to pick one based on it's characteristics (rather than intuitively knowing what the characteristics of the class are intuitively). I'm not sure how necessary this is unless you have more than 5 Gateways available (which I feel is rare btw idk).
- Binding via Gateway symbolic reference like `prod-web-gw=true` or `prod-internal-gw=true` would allow a Route to be exposed by multiple Gateways.

Reply:
@mark-church Thanks for the summary of potential approaches here! I've added something to the agenda for the community meeting tomorrow. I think it will be helpful to have some broader discussion around this.

Reply:
changed this to `expose=prod-web-gw` for the simplest approach that does not overload use of the term gateway